walkthroughs/ec2/README.md

```diff
@@ -0,0 +1,149 @@
+# Using AWS App Mesh with EC2
+
+## Overview
+
+In this article, I demonstrate running a service on EC2 and configuring to run with an existing ECS application, adding a route for the service using AWS App Mesh.
+
+In previous articles I gave walkthroughs for running ECS and Fargate services for a demo application called the Color App using AWS App Mesh to control load balancing and routing (see [ECS walkthrough], [Fargate walkthrough]), so in this article I will just highlight the additional steps for adding the another version of this service that will run on an EC2 instance.
+
+If you haven't read the previous articles, you should at least start with the [ECS walkthrough] first and ensure you have the application up and running. Once you are able to query for colors ([initially load balanced equally between a few colors](https://github.com/aws/aws-app-mesh-examples/blob/master/examples/apps/colorapp/servicemesh/appmesh-colorapp.yaml#L127-L132), then you'll be ready to follow along with this walkthrough.
+
+## Prerequisites
+
+1. You have successfully set up the prerequisites and deployed the Color App as described in the previous [ECS walkthrough].
+
+## Deploy
+
+In a clone of the [repo], ensure your shell is updated with the environment variables described in the [walkthrough].
+
+As a reminder for the environment variables should look lie, here's what I'm using for the demo:
+
+```
+export AWS_PROFILE=default
+export AWS_DEFAULT_REGION=us-west-1
+export AWS_REGION=$AWS_DEFAULT_REGION
+export ENVIRONMENT_NAME=demo
+export MESH_NAME=colorapp
+export SERVICES_DOMAIN=demo.local
+export ENVOY_IMAGE=111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.9.1.0-prod
+export KEY_PAIR_NAME="YOUR IAM USER KEY PAIR"
+```
+
+### 1. Deploy the App Mesh configuration update
+
+We will update the existing mesh for the application by adding support for a yellow colorteller.
+
+In `walkthroughs/ec2/appmesh-colorapp.yaml`, we've added a resource for the virtual node (`ColorTellerYellowVirtualNode`) that will ultimately get mapped to a service that we'll run on our EC2 instance listening for HTTP requests on port 9080. This virtual node will be registered to be advertised for service discovery.
```

Comment:
So is the customer expected to tear down their mesh from the fargate walkthrough? or is this going to run on top of their previous one?

i wonder if there is a nice way to define the mesh in a central place and opt-in different platforms? This would be a nice example for customers on how to deploy a few different targets in the same account.

Reply:
It runs on top of and can be torn down separately, like the Fargate walkthrough as well. These do mention that they build on top of the ECS demo. I do plan on an update that also covers the migration case of migrating from conventional EC2 to managed container services where you're using App Mesh to aid in decomposing to microservices without breaking existing clients.